### DISCUSSION TITLE
Setting Redaction Code vs. Custom Text

### DISCUSSION THREAD

**Discussion by mwade-noetic** [pymupdf/PyMuPDF#1570] [Looking for help]
I am creating redactions, but not burning them into, they show up as crossed out text.  In Acrobat Pro, in the Redaction properties there are two properties that contain the text the will be placed into the redaction:  "Custom Text" and "Redaction Code".   The screen shot below is an example of that.

![image](https://user-images.githubusercontent.com/79216072/151643830-53eff0bf-a25f-4f96-be53-1b5b13128820.png)

I cannot figure out how to set the "Redaction Code" vs. the Custom Text.  Using the standard way of setting the redaction box text, it shows up this way.  Is there a way to set it as "redaction code" instead?  This would allow the users to review and accept the annotations with less work.

Currently I am calling "add_redact_annot() and setting text="redaction text".   Perhaps I do this with a custom entry in the annotation info dictionary?  I am setting the "content" to the redaction code as well there.  

Thanks for any help.

**Comment by JorjMcKie** [comment]
I don't knows what Adobe means by **_redaction code_**. It is not part of the official set of dictionary keys of a redaction annotation.
In PyMuPDF, the "text" parameter is what will be put in the redact rectangle when applying, and therefore is the OverlayText.
You could look at the redaction definition after Adobe has been fiddling with it `print(doc.xref_object(annot.xref))` and inspect where this ominous data has landed.

The problem here is like alays:
Annotations and fields are tool-specific when it comes to the technical implemementation, because there do not exist any rules that must be obeyed.
This implies:
* An annot made with some other tool, will look **_completely_** different after being updated with PyMuPDF. In addition, all original features are removed from the annot which PyMuPDF does not support.
* The opposite is also true: make an annot with PyMuPDF, then update it with e.g. Adobe and you will see drastic changes happening.

**Comment by mwade-noetic** [comment]
Thanks,

I just realized in the last few minutes that they are importing an external set of codes that you can “Select” from a drop-down list below (in the screenshot) and use that to populate a code in the annotation.  I thought they were pulling it from the annotation information itself when I first saw it.  So in this case, they are just overlaying the annotation information with this custom external data that is external to the PDF itself.

Thanks for the fast response!

Michael Wade
